.azure-pipelines/auto.yml

```
@@ -0,0 +1,356 @@
+#      x86_64-gnu-debug:
+#        IMAGE: x86_64-gnu-debug
+#      x86_64-gnu-nopt:
+#        IMAGE: x86_64-gnu-nopt
+#      x86_64-gnu-distcheck:
+#        IMAGE: x86_64-gnu-distcheck
+      mingw-check:
+        IMAGE: mingw-check
+
+- job: macOS
+  timeoutInMinutes: 180
+  pool:
+    vmImage: macos-10.13
+  steps:
+  - checkout: self
+    fetchDepth: 2
+  - template: steps/macos.yml
+  strategy:
+    matrix:
+      # macOS builders. These are placed near the beginning because they are very
+      # slow to run.
+
+      # OSX builders running tests, these run the full test suite.
+      # NO_DEBUG_ASSERTIONS=1 to make them go faster, but also do have some
+      # runners that run `//ignore-debug` tests.
+      #
+      # Note that the compiler is compiled to target 10.8 here because the Xcode
+      # version that we're using, 8.2, cannot compile LLVM for OSX 10.7.
+      x86_64-apple:
+        RUST_CHECK_TARGET: check
+        RUST_CONFIGURE_ARGS: --build=x86_64-apple-darwin --enable-sanitizers --enable-profiler --set rust.jemalloc
+        RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
+        MACOSX_DEPLOYMENT_TARGET: 10.8
+        MACOSX_STD_DEPLOYMENT_TARGET: 10.7
+        NO_LLVM_ASSERTIONS: 1
+        NO_DEBUG_ASSERTIONS: 1
+
+      dist-x86_64-apple:
+        RUST_CHECK_TARGET: dist
+        RUST_CONFIGURE_ARGS: --target=aarch64-apple-ios,armv7-apple-ios,armv7s-apple-ios,i386-apple-ios,x86_64-apple-ios --enable-full-tools --enable-sanitizers --enable-profiler --enable-lldb --set rust.jemalloc
+        DEPLOY: 1
+        RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
+        MACOSX_DEPLOYMENT_TARGET: 10.7
+        NO_LLVM_ASSERTIONS: 1
+        NO_DEBUG_ASSERTIONS: 1
+        DIST_REQUIRE_ALL_TOOLS: 1
+
+#      dist-x86_64-apple-alt:
+#        RUST_CHECK_TARGET: dist
+#        RUST_CONFIGURE_ARGS: --enable-extended --enable-profiler --enable-lldb --set rust.jemalloc
+#        DEPLOY_ALT: 1
+#        RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
+#        MACOSX_DEPLOYMENT_TARGET: 10.7
+#        NO_LLVM_ASSERTIONS: 1
+#        NO_DEBUG_ASSERTIONS: 1
+#
+#      i686-apple:
+#        RUST_CHECK_TARGET: check
+#        RUST_CONFIGURE_ARGS: --build=i686-apple-darwin --set rust.jemalloc
+#        RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
+#        MACOSX_DEPLOYMENT_TARGET: 10.8
+#        MACOSX_STD_DEPLOYMENT_TARGET: 10.7
+#        NO_LLVM_ASSERTIONS: 1
+#        NO_DEBUG_ASSERTIONS: 1
+#
+#      dist-i686-apple:
+#        RUST_CHECK_TARGET: dist
+#        RUST_CONFIGURE_ARGS: --build=i686-apple-darwin --enable-full-tools --enable-profiler --enable-lldb --set rust.jemalloc
+#        DEPLOY: 1
+#        RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
+#        MACOSX_DEPLOYMENT_TARGET: 10.7
+#        NO_LLVM_ASSERTIONS: 1
+#        NO_DEBUG_ASSERTIONS: 1
+#        DIST_REQUIRE_ALL_TOOLS: 1
+
+
+
+- job: Windows
+  timeoutInMinutes: 180
+  pool:
+    vmImage: 'vs2017-win2016'
```

as this is an evaluation maybe use vs2019.
when I have tested on my computer I have seen 7% faster build of rustc from a clean repo
```suggestion
    vmImage: 'windows-2019'
```

---

Hmm, I'd like to keep an environment as similar as the current Travis/AppVeyor one at least for the time being. We'll surely investigating a new Visual Studio version in the future though, thanks for pointing this out!